### ISSUE TITLE
Filament Temperature Offset not working

### ISSUE THREAD

**Issue by enderjones** [OctoFarm/OctoFarm#672]
### OctoFarm Version

1.1.13-hotfix

### OctoPrint Version(s)

1.5.3

### pm2 or docker?

pm2

### Operating System

Ubuntu via Hyper-V

### OctoPrint Plugins

SMS notification

### Location of the problem

Not sure

### Description of the problem

```bash
I tried to put a -20 degC offset on one of my spools of filament, but it did not offset when I started the print.
```

### Steps taken to reproduce

```bash
1.
2.
3.
...
```

### What did you expect to happen

```bash
The temperature to be reduced by 20 degC
```

### To avoid the problem I had to

```bash
...
```

**Comment by davidzwa** [comment]
Please provide an exact OctoFarm version because 'latest' is ambiguous as the 1.2 release candidate is published.

**Comment by enderjones** [comment]
@davidzwa updated the version.

**Comment by NotExpectedYet** [comment]
@davidzwa Its client side so unless there's any console logs happening when starting a print I'll just have to repro at minute. I know where the code is at a guess it's something not accounting for a (-) figure as this was fixed and tested a bit ago but only remember testing positives. 

Thanks for the report @enderjones I'll take a look and see if I can't get it fixed for rc2. Would you be able to test if so?

**Comment by enderjones** [comment]
@NotExpectedYet Yeah for sure.  Thanks for looking into it!

**Comment by enderjones** [comment]
Odd observation.  I started the print for the 3rd time and the temperature offset worked fine...  I did not change anything, just started the same file and it worked correct

**Comment by NotExpectedYet** [comment]
Hmm well, I'll leave this open till I can actually investigate it more. I think I have an idea why anyway. 

I forget the conditions that are in that section of code but there's a fair few from tacking on features over the course of last year. I gather you managed to hit a condition I didn't update when testing it last time. More reason to clean that code up :grinning:

If you have any info to the steps you took and what might be different will help account to it in the future. 

Least it's now working

**Comment by pat1121** [comment]
Just to add to this, I can confirm that the temperature offset:

1) Does NOT work if you print from the file manager (uses standard temps in Gcode)
2) DOES work if you print by selecting the printer in the printer menu, then selecting the file you'd like to print

I am using a +20 temp offset, haven't tried negatives.

**Comment by NotExpectedYet** [comment]
> Just to add to this, I can confirm that the temperature offset:
> 
> 1. DOES work if you print from the file manager (uses standard temps in Gcode)
> 2. DOES work if you print by selecting the printer in the printer menu, then selecting the file you'd like to print
> 
> It was suggested it only worked from the printer menu, I got it to work from both.
> I am using a +20 temp offset, haven't tried negatives.
> 
> It did NOT work the first time I tried to print the file (on a few separate printers). I had to cancel (before it even started) the first print, then tell it to print again. On the second attempt, the temperature offset worked. This happened on 4 printers, 4 separate prints the exact same way, using either the printer menu or the file menu to start the print.

Thanks for that! Makes it easier to track down :)

**Comment by pat1121** [comment]
On the second print (using the same printers after going through the start / cancel procedure above) everything works as intended using all 3 ways to print (reprint, printer menu, file menu) without having to do the start / cancel part.

**Comment by NotExpectedYet** [comment]
Sorry about my tardiness with this issue but I've got it solved for v1.2. 

Works from any area in the UI with positive and minus values.

Normal: 
![image](https://user-images.githubusercontent.com/61137988/137626763-574933f7-1dd6-4000-b32b-c108f9cbf2fc.png)
-25: 
![image](https://user-images.githubusercontent.com/61137988/137626787-f12f4528-a306-4fe2-90f1-d35b2413b3f6.png)
+25:
![image](https://user-images.githubusercontent.com/61137988/137626817-de41ac9b-8993-4ce9-bf57-e3d93699d6f8.png)

It worked after the second one because I was incorrectly sending the offset after the start print command, so the offset had been applied by that time. Will update here once the fix has hit dev / master.
